`make container-build` command fails

**Problem:**

Run the following command:

```console
$ make container-build
"docker" run --rm --interactive --tty --volume "/Users/chalin/git/lf/kubernetes/kubernetes.io:/src:ro,Z" --read-only --mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.133.0-663a78832a43 sh -c "npm ci && hugo --minify --environment development"
npm error code EROFS
npm error syscall unlink
npm error path /src/node_modules/.package-lock.json
npm error errno -30
npm error rofs EROFS: read-only file system, unlink '/src/node_modules/.package-lock.json'
npm error rofs Often virtualized file systems, or other file systems
npm error rofs that don't support symlinks, give this error.
npm error Log files were not written due to an error writing to the directory: /var/hugo/.npm/_logs
npm error You can rerun the command with `--loglevel=verbose` to see the logs in your terminal
make: *** [container-build] Error 226
```

Note the errors reported by NPM.

**Analysis:**

The issue is that `/src` is mounted as read-only, and so we can't run `npm ci` as is from within the container.

This issue is reproducible (with slightly different messages in my env; e.g., `EROFS` vs. `ENOENT`):
```
$ make container-build
"docker" run --rm --interactive --tty --volume "/home/jhseo/website:/src:ro,Z" --read-only --mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.133.0-d50c05f50dfc sh -c "npm ci && hugo --minify --environment development"
npm error code ENOENT
npm error syscall mkdir
npm error path /src/node_modules
npm error errno -2
npm error enoent ENOENT: no such file or directory, mkdir '/src/node_modules'
npm error enoent This is related to npm not being able to find a file.
npm error enoent
npm error Log files were not written due to an error writing to the directory: /var/hugo/.npm/_logs
npm error You can rerun the command with `--loglevel=verbose` to see the logs in your terminal
make: *** [Makefile:100: container-build] Error 254
```

My env:
- Ubuntu 22.04
- Commit dea6799 @ branch `main`

--- a/.dockerignore
+++ b/.dockerignore
@@ -1,1 +1,3 @@
 **
+!package.json
+!package-lock.json

--- a/Dockerfile
+++ b/Dockerfile
@@ -31,8 +31,7 @@ RUN apk add --no-cache \
     git \
     openssh-client \
     rsync \
-    npm && \
-    npm install -D autoprefixer postcss-cli
+    npm
 
 RUN mkdir -p /var/hugo && \
     addgroup -Sg 1000 hugo && \
@@ -43,6 +42,8 @@ RUN mkdir -p /var/hugo && \
 COPY --from=0 /go/bin/hugo /usr/local/bin/hugo
 
 WORKDIR /src
+COPY package.json package-lock.json ./
+RUN npm ci
 
 USER hugo:hugo
 

--- a/Makefile
+++ b/Makefile
@@ -97,11 +97,17 @@ docker-push: ## Build a multi-architecture image and push that into the registry
 	rm Dockerfile.cross
 
 container-build: module-check
-	$(CONTAINER_RUN) --read-only --mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 $(CONTAINER_IMAGE) sh -c "npm ci && hugo --minify --environment development"
+	mkdir -p public
+	$(CONTAINER_RUN) --read-only \
+		--mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 \
+		--mount type=bind,source=$(CURDIR)/public,target=/src/public $(CONTAINER_IMAGE) \
+		hugo --cleanDestinationDir --buildDrafts --buildFuture --environment preview --noBuildLock
 
 # no build lock to allow for read-only mounts
 container-serve: module-check ## Boot the development server using container.
-	$(CONTAINER_RUN) --cap-drop=ALL --cap-add=AUDIT_WRITE --read-only --mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 -p 1313:1313 $(CONTAINER_IMAGE) hugo server --buildFuture --environment development --bind 0.0.0.0 --destination /tmp/hugo --cleanDestinationDir --noBuildLock
+	$(CONTAINER_RUN) --cap-drop=ALL --cap-add=AUDIT_WRITE --read-only \
+		--mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 -p 1313:1313 $(CONTAINER_IMAGE) \
+		hugo server --buildFuture --environment development --bind 0.0.0.0 --destination /tmp/public --cleanDestinationDir --noBuildLock
 
 test-examples:
 	scripts/test_examples.sh install

